tenant_schemas_celery/task_test.py

```diff
@@ -3,10 +3,15 @@
 from freezegun import freeze_time
 
 from tenant_schemas_celery.task import TenantTask
+from tenant_schemas_celery.app import CeleryApp
 from tenant_schemas_celery.test_utils import create_client
-
+from django.conf import settings
 
 def test_task_get_tenant_for_schema_should_cache_results(transactional_db):
+    # Celery is not able to pick up settings without this
```

Comment:
Could you please split this test into more separated units? Currently it checks two completely separate things:

* per-task caching
* global caching

These things are completely different features and should have separate tests.

Another thing you don't test is global caching with two different tasks - the tenant objects should be re-used in such case, no?

Reply:
One thing this combined test shows that separate tests don't is that the local setting overwrites the global one.

I could create a separate test but how should that scenario be covered?

Reply:
> One thing this combined test shows that separate tests don't is that the local setting overwrites the global one.
> 
> I could create a separate test but how should that scenario be covered?

On other hand as long as the per-task caching test case has a TTL setting less than global, we know the overwrite works. So have split out the test cases and added to missing one as suggested